schedule/scheduler.go

```diff
@@ -91,9 +94,15 @@ func (s *Scheduler) Run(ctx context.Context) error {
 				if err := tc.Locker.Lock(ctx); err != nil {
 					if errors.Is(err, storage.ErrLockNotAcquired) {
 						log.Infow("task not started: lock not acquired", "task", tc.Name)
+						if tc.ExitOnFailure {
+							log.Fatal("Exiting due to previous error. ExitOnFailure")
```

Comment:
I'm not sure this is the behavior we want:
```
// Fatal uses fmt.Sprint to construct and log a message, then calls os.Exit.
```
https://github.com/uber-go/zap/blob/master/sugar.go#L126

None of the deferred unlocking code will be called if we exit here, and per my above comment I don't think the `ExitOnFailure` flag is needed.

Reply:
Strong agree. Using os.Exit almost guarantees that any lock we manage to take will be left active in the database. So in some circumstances this exit behaviour will make our problem worse.

Reply:
As discussed, agreed to remove here.